api/src/main/java/jakarta/data/repository/CrudRepository.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022,2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.repository;
+
+import jakarta.data.exceptions.EntityExistsException;
+
+/**
+ * <p>A repository interface that extends the capabilities of basic operations on entities, including insert and update operations.</p>
+ *
+ * <p>This repository extends the {@link BasicRepository} interface, providing a comprehensive set of methods to interact with
+ * persistent entities of type {@code <T>}, where {@code <T>} represents the entity bean type, and {@code <K>} represents the key type.</p>
+ *
+ * <p>It encompasses standard CRUD (Create, Read, Update, Delete) operations, allowing you to perform insert and update operations in
+ * addition to basic retrieval and deletion. This interface combines the Data Access Object (DAO) aspect with the repository pattern,
+ * offering a versatile and complete solution for managing persistent entities within your Java applications.</p>
+ *
+ * @param <T> the entity bean type
+ * @param <K> the key type.
+ * @see BasicRepository
+ * @see DataRepository
+ */
+public interface CrudRepository<T, K> extends BasicRepository<T, K> {
+
+    /**
+     * <p>Inserts an entity into the database. If an entity of this type with the same
+     * unique identifier already exists in the database, then this method raises
+     * {@link EntityExistsException} for.</p>
+     *
+     * @param entity the entity to insert. Must not be {@code null}.
+     * @return the inserted entity.
+     * @throws EntityExistsException if the entity is already present in the database.
+     * @throws NullPointerException if the entity is null.
+     * @throws UnsupportedOperationException for databases that use an append model to write data.
+     */
+    T insert(T entity);
```

Comment:
@njr-11 is it fine to SQL, isn't it?

In NoSQL we can work with record, so, return the void, we will lost the link. That is why I would like this return.

```java
record Book(UUID databaseGenerated, String name){}

var newBook= repository.insert(book);
```

Reply:
> @njr-11 is it fine to SQL, isn't it?
> 
> In NoSQL we can work with record, so, return the void, we will lost the link. That is why I would like this return.
> 
> ```java
> record Book(UUID databaseGenerated, String name){}
> 
> var newBook= repository.insert(book);
> ```

There isn't any difference between record and other types of entities here because Jakarta Data doesn't require or in any way even suggest that an entity that is supplied as a parameter will be updated.  There also isn't any difference between relational and NoSQL here.  Whenever you want the pattern of being able know the state of thing you are inserting or update, you use `save`, which provides that ability, not `insert`.  Inserts are not supposed to be queries that fetch information.